vm/src/obj/objtype.rs

```diff
@@ -26,19 +26,35 @@ impl PyValue for PyClass {
     }
 }
 
+impl IdProtocol for PyClassRef {
+    fn get_id(&self) -> usize {
+        self.as_object().get_id()
+    }
+
+    fn is(&self, other: &Self) -> bool {
+        self.get_id() == other.get_id()
+    }
+}
+
+impl TypeProtocol for PyClassRef {
+    fn type_ref(&self) -> &PyObjectRef {
+        &self.as_object().type_ref()
+    }
+}
+
 struct IterMro<'a> {
     cls: &'a PyClassRef,
     offset: Option<usize>,
 }
 
 impl<'a> Iterator for IterMro<'a> {
-    type Item = &'a PyObjectRef;
+    type Item = &'a PyClassRef;
```

Comment:
What's the reason for holding `PyClassRef` by reference? Is it just to avoid an extra inc and dec of the ref count?

Reply:
Yes, pretty much.

It wasn't a very deliberate decision. My initial work here tended towards just clone everything all the time, and now I'm trying to just take references whenever I can.

Reply:
Not a big deal, but personally I'd prefer either `&PyClass` or to just clone the `PyClassRef` (which is pretty cheap anyway). Shared refs to reference counted types just seem kind of weird.